chore: Auto generate colors via SwiftGen 🎨

abdelhamid-f-nasser opened the pull request:
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature

## Description
- Adds generated 🎨 UI-Colors using  🧬swiftGen .
- Note: Moved the colors to a separate assets bundle.
- Add all default colors 🎨
- Add system colors ( to expose the colors to our components for better readability)

## Related Tickets & Documents
Cannot access Jira to find ticket number
<!-- 
Please use this format link issue numbers: Fixes #123
https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword 
-->
## Added tests?
- [x] 🙋 no, because I need help

## Added to documentation?
- [x] 🙅 no documentation needed


## [optional] What gif best describes this PR or how it makes you feel?
![Robot fails](https://media.giphy.com/media/12yMPFcQ6FQF8Y/giphy.gif)

abdelhamid-f-nasser committed:
chore(Colors): Auto generate colors via SwiftGen 🎨

- Moved the colors to a separate assets bundle.
- Generate colors using SwiiftGen

abdelhamid-f-nasser committed:
chore(Colors): Auto generate colors via SwiftGen 🎨

- Moved the colors to a separate assets bundle.
- Generate colors using SwiiftGen

abdelhamid-f-nasser committed:
style: Fix linting violations 💇🏼‍♂️

- resolve linting violation in `SceneDelegate.swift` and `HealthyValidationRule.swift`

Ahattalla committed:
feat: [HL-18] Add and configure Xcode template 🔨 (#22)

* Added MVVM template, and makefile that automatically add the template to xcode and to remove it

* Fix swiftlint conflict with MVVM-Template files by excluding them

* Removing __FILEHEADER__ identifier to match linting rules of not having a header

* Removed dead code and rename MVVM-Template to Templates

abdelhamid-f-nasser committed:
chore(Colors): Add all default colors 🎨

Added all default colors for the application

abdelhamid-f-nasser committed:
chore(Colors): Add all default colors 🎨

Added all default colors for the application

abdelhamid-f-nasser committed:
style(Generated): Reformat UIImage generated file 💇🏼‍♂️

abdelhamid-f-nasser committed:
style: Fix linting violations 💇🏼‍♂️

- resolve linting violation in `SceneDelegate.swift` and `HealthyValidationRule.swift`

abdelhamid-f-nasser committed:
style: Fix linting violations 💇🏼‍♂️

- resolve linting violation in `SceneDelegate.swift` and `HealthyValidationRule.swift`

abdelhamid-f-nasser committed:
style(Generated): Reformat UIImage generated file 💇🏼‍♂️

abdelhamid-f-nasser committed:
style(Generated): Reformat UIImage generated file 💇🏼‍♂️

ahmdmhasn commented on Healthy/Classes/Generated/UIColors+Generated.swift:
```diff
@@ -0,0 +1,89 @@
+// swiftlint:disable force_unwrapping
+extension UIColor {
+
+      static var black: UIColor {
+        UIColor(named: "Black")!
+      }
+
+      static var gray1: UIColor {
+        UIColor(named: "Gray 1")!
+      }
+
+      static var gray2: UIColor {
+        UIColor(named: "Gray 2")!
+      }
+
+      static var gray3: UIColor {
+        UIColor(named: "Gray 3")!
+      }
+
+      static var gray4: UIColor {
+        UIColor(named: "Gray 4")!
+      }
```
```suggestion
      static var black: UIColor {
        UIColor(named: "Black")!
      }

      static var gray1: UIColor {
        UIColor(named: "Gray 1")!
      }

      static var gray2: UIColor {
        UIColor(named: "Gray 2")!
      }

      static var gray3: UIColor {
        UIColor(named: "Gray 3")!
      }

      static var gray4: UIColor {
        UIColor(named: "Gray 4")!
      }
```

Let's update black to have the same color incremental values. ex. 20-40-60-80-100(Pure Black)

abdelhamid-f-nasser replied:
Just to be sure, Should we add shades to the black from black 20 to black 100? and should I also do this to the rest of the colors?

abdelhamid-f-nasser replied:
Oh so we change the grays to black 20-80 and the current black to Black 100?

ahmdmhasn replied:
Yes exactly, just the black.

ahmdmhasn reviewed (approved):
🚀

Ahattalla reviewed (approved):
Good job ya bro ❤️

abdelhamid-f-nasser committed:
refactor(Colors): Update black and grey colors to shades of black 🔨

Refactor the color scheme by renaming the black and gray shades. The previous names, which were "black" and "gray 1," "gray 2," "gray 3," "gray 4,"..

They have been replaced with more descriptive names: "black 100," "black 80," "black 60," "black 40," and "black 20."

The actual colors themselves remain unchanged. This update enhances clarity and consistency within the color palette.

ahmdmhasn commented:
@abdelhamid-f-nasser Please make sure to drop all the irrelevant changes.

ahmdmhasn committed:
Fix rebase issues

ahmdmhasn committed:
Merge branch 'develop' into feat/auto-generate-colors-using-swiftgen

# Conflicts:
#	.swiftlint.yml
#	Healthy.xcodeproj/project.pbxproj
#	Healthy/Classes/SceneDelegate.swift
